torch/csrc/jit/custom_class.cpp

```diff
@@ -21,20 +22,127 @@ bool isCustomClass(const c10::IValue& v) {
       getCustomClass(v.toObject()->type()->name()->qualifiedName());
 }
 
+at::TypePtr getCustomClass(const std::string& name) {
+  return classCU()->get_type(name);
+}
+
 namespace {
 
-TypePtr realCustomClassHandler(const std::string& name) {
-  return classCU()->get_type(name);
+std::unordered_map<const detail::RegisteredClassRecord*, ClassTypePtr> &registeredClassTypes() {
+  static std::unordered_map<const detail::RegisteredClassRecord*, ClassTypePtr> table;
+  return table;
+}
+
+std::set<ClassTypePtr> &getstateSetstateValidated() {
+  static std::set<ClassTypePtr> s;
+  return s;
+}
+
+void classCallback(const detail::RegisteredClassRecord& clazz) {
+  auto classTypePtr =
+      ClassType::create(c10::QualifiedName(clazz.qualClassName), classCU());
```

Comment:
do we need ClassType to exist for IValue to work? not really sure how capsule type works